markets/perps-market/contracts/storage/GlobalPerpsMarket.sol

```diff
@@ -63,6 +65,30 @@ library GlobalPerpsMarket {
         }
     }
 
+    function utilizationRate(
+        Data storage self
+    ) internal view returns (uint128 rate, uint256 delegatedCollateralValue, uint256 lockedCredit) {
+        uint256 withdrawableUsd = PerpsMarketFactory.totalWithdrawableUsd();
+        delegatedCollateralValue = withdrawableUsd - totalCollateralValue(self);
```

Comment:
I was planning to use `getMarketCollateral`
```
    /**
     * @notice Returns the total snxUSD value of the collateral for the specified market.
     * @param marketId The id of the market whose collateral is being queried.
     * @return valueD18 The market's total snxUSD value of collateral, denominated with 18 decimals of precision.
     */
    function getMarketCollateral(uint128 marketId) external view returns (uint256 valueD18);
```
Is there a reason that shouldn't be used?

Reply:
`getMarketCollateral` is the value of collateral deposited by the market, not the collateral delegated to it.  maybe in your case, you can use this to get the `totalCollateralValue` but you will want to subtract that out from whatever `getWithdrawableUsd` returns which includes everything including whatever market has deposited.

Reply:
hmm I think getMarketCollateral is the delegated collateral from pools.
Core have `getMarketCollateralValue` and `getMarketCollateralAmount` which return the collateral deposited by the maket

Reply:
```
 function getMarketCollateral(uint128 marketId) external view override returns (uint256) {
        return Market.load(marketId).poolsDebtDistribution.totalSharesD18;
    }
```
I'm not totally sure how it works, but it's getting the data from pool's debt distribution

Reply:
I think doing it they way you do it:`withdrawableUsd - totalCollateralValue(self);` means we not only get delegated collateral since the `withdrawableUsd` also takes the market's creditCapacity into account. 
Maybe we want that?

It basically means that if the market has a bunch of unrealised gains the the utilisation will go down, as the credit capacity takes those that into account through the reportedDebt

Reply:
We do want creditCapacity included so this is correct